test/integration/dbversion.test.js

```diff
@@ -0,0 +1,45 @@
+describe('User service integration test', () => {
+  const dbVersion = require('../../server/dbversion')
+
+  async function truncateVersionTable () {
+    return dbVersion.umzug.storage.model.destroy({
+      where: {},
+      truncate: true
+    })
+  }
+
+  beforeEach(async () => {
+    await truncateVersionTable()
+  })
+
+  test('dbVersion gets a list of available migrations', () => {
+    expect(dbVersion.availableVersions.length).toBeGreaterThan(0)
+  })
+
+  test('dbVersion validates version in database', async () => {
+    await dbVersion.refreshCurrentDatabaseVersion()
+    expect(dbVersion.currentDatabaseVersion).not.toBe(dbVersion.highestVersion)
```

Comment:
As far as I can tell, the database table will be empty (it's truncated in `beforeEach`) so refreshing the current version will achieve nothing, regardless of whether it works or not.

Do we need to insert some data into the table before calling `refreshCurrentDatabaseVersion`, then assert that `dbVersion.currentDatabaseVersion` matches the last version inserted?

Reply:
Added a database insert and check against that version.

Reply:
I've removed the unneeded expect.assertions() and tidied up the tests around exceptions, to make sure that the exceptions returned are what's expected by the tests, and a further test to check all exception paths with the necessary mocking.